cocoa/src/toga_cocoa/window.py

```diff
@@ -288,10 +294,57 @@ def get_visible(self):
     # Window state
     ######################################################################
 
-    def set_full_screen(self, is_full_screen):
-        current_state = bool(self.native.styleMask & NSWindowStyleMask.FullScreen)
-        if is_full_screen != current_state:
-            self.native.toggleFullScreen(self.native)
+    def get_window_state(self):
+        if self.interface.content and bool(
+            self.interface.content._impl.native.isInFullScreenMode()
+        ):
+            return WindowState.PRESENTATION
+        elif bool(self.native.styleMask & NSWindowStyleMask.FullScreen):
+            return WindowState.FULLSCREEN
+        elif bool(self.native.isZoomed):
+            return WindowState.MAXIMIZED
+        elif bool(self.native.isMiniaturized):
+            return WindowState.MINIMIZED
+        else:
+            return WindowState.NORMAL
+
+    def set_window_state(self, state):
+        current_state = self.get_window_state()
+        if state == WindowState.NORMAL:
+            # If the window is maximized, restore it to its normal size
+            if current_state == WindowState.MAXIMIZED:
+                self.native.setIsZoomed(False)
+            # Deminiaturize the window to restore it to its previous state
+            elif current_state == WindowState.MINIMIZED:
+                self.native.setIsMiniaturized(False)
+            # If the window is in full-screen mode, exit full-screen mode
+            elif current_state == WindowState.FULLSCREEN:
+                # This doesn't wait for completely exiting full screen mode. Hence,
+                # this will cause problems when rapid window state switching is done.
+                # We should wait until `windowDidExitFullScreen_` is notified and then
+                # return to user.
+                self.native.toggleFullScreen(self.native)
+            # If the window is in presentation mode, exit presentation mode
+            elif current_state == WindowState.PRESENTATION:
+                self.interface.app.exit_presentation_mode()
+        else:
+            if state == WindowState.MAXIMIZED:
+                self.native.setIsZoomed(True)
+
+            elif state == WindowState.MINIMIZED:
+                self.native.setIsMiniaturized(True)
+
+            elif state == WindowState.FULLSCREEN:
+                self.native.toggleFullScreen(self.native)
+
+            elif state == WindowState.PRESENTATION:
+                self.interface.app.enter_presentation_mode(
+                    {self.interface.screen: self.interface}
+                )
+            else:  # pragma: no cover
```

Comment:
Changed it.

Reply:
Sure - but you've left the `#pragma:no cover` in place.

Reply:
>Sure - but you've left the #pragma:no cover in place.

But, since that else branch would never be touched, so removing `#pragma:no cover` would give missing coverage.

Reply:
But, `exit_presentation_mode()` is called early in `window.state`, so if the window was in presentation mode then it would have exited presentation mode by the time `set_presentation_mode()` would be called. So, `current_state` will never be PRESENTATION when checked inside `_apply_state()`. Hence, this branch cannot be reached.